### PR TITLE
[chore] Prepare release v0.106.0

### PR DESCRIPTION
**Pull request by songy23**

### CODE REVIEW

**Review by crobert-1 — APPROVED** [review]

**Review by mwear — APPROVED** [review]

File: cmd/githubgen/go.mod [hunk 1]
```diff
@@ -19,7 +19,7 @@ require (
 	github.com/mitchellh/copystructure v1.2.0 // indirect
 	github.com/mitchellh/reflectwalk v1.0.2 // indirect
 	go.opentelemetry.io/collector/featuregate v1.12.1-0.20240729082905-fb5b1e6aa550 // indirect
```
Comment (codeboten) [inline cmd/githubgen/go.mod:21]: The stable modules should be set to a released version

Comment (songy23) [reply cmd/githubgen/go.mod:21]: It was v0.12.0 before, then there are multiple commits bumping the version. The latest is https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/9128a9258fe1fee36f198f97b1e3371fc7b77a93.

I can try revert it back to v0.12.0

Comment (codeboten) [reply cmd/githubgen/go.mod]: i'm assuming you mean 1.12 :) in which case yeah that makes sense

**Review by codeboten — APPROVED** [review]

**Comment by songy23** [comment]
should be ready to merge 🙏

**Review by atoulme — APPROVED** [review]

**Pull request merged**